keys/signer_secp256r1_test.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2020 Coinbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+package keys
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/coinbase/rosetta-sdk-go/types"
+)
+
+var signerSecp256r1 Signer
+
+func init() {
+	keypair, _ := GenerateKeypair(types.Secp256r1)
+	signerSecp256r1, _ = keypair.Signer()
+}
+
+func TestSignSecp256r1(t *testing.T) {
+	type payloadTest struct {
+		payload *types.SigningPayload
+		sigType types.SignatureType
+		sigLen  int
+		err     bool
+		errMsg  error
+	}
+
+	var payloadTests = []payloadTest{
+		{mockPayload(hash("hello123"), types.Ecdsa), types.Ecdsa, 64, false, nil},
+		{
+			mockPayload(hash("hello1234"), types.EcdsaRecovery),
+			types.EcdsaRecovery,
+			65,
+			true,
+			ErrSignUnsupportedSignatureType,
```

Comment:
Is ECDSA Recovery truly not supported for secp256r1? Interesting.

Reply:
This means `type.EcdsaRecovery` sigType for secp256r1 is unsupported for rosetta-go-sdk, not `secp256r1`. ECDSA recovery **IS** supported for `secp256r1`, but his PR would not support this for some reasons:

1. Golang's `crypto/ecdsa` package doesn't provide a `SignRecoverable`  method for `secp256r1` to create recoverable signature({r, s, v}, v is the recovery id). And I didn't find some other long-tested packages provide this. 

2. I know ONT, NEO, ZEEPIN use normal `secp256r1` signatures in their transactions, but I don't know the use case of `secp256r1` recoverable signature. It seems many people just recover the PublicKey from {r, s}, for v is not necessary strictly speaking.

Reply:
Ok, good to know. Better to leave something out if not really used. Less code is more!